pymemcache/client/retrying.py

```diff
@@ -0,0 +1,178 @@
+""" Module containing the RetryingClient wrapper class. """
+
+from time import sleep
+
+
+def _ensure_tuple_argument(argument_name, argument):
+    """
+    Helper function to ensure the given arguments are tuples of Exceptions (or
+    subclasses), or can at least be converted to such.
+
+    Args:
+      argument_name: str, name of the argument we're checking, only used for
+        raising meaningful exceptions.
+      argument: any, the argument itself.
+
+    Returns:
+      tuple[Exception]: A tuple with the elements from the argument if they are
+        valid.
+
+    Exceptions:
+      ValueError: If the argument was not None, tuple or Iterable.
+      ValueError: If any of the elements of the argument is not a subclass of
+        Exception.
+    """
+
+    # Ensure the argument is a tuple, set or list.
+    if argument is None:
+        return tuple()
+    elif not isinstance(argument, (tuple, set, list)):
+        raise ValueError("%s was not a tuple or Iterable." % argument_name)
```

Comment:
Sorry I didn't noticed this exception message during my previous review.

What do you think about a message like this:

> An error occur with your passed argument. This method accept three possible kind of sequence type (tuple, set, list) for the argument "argument". If you see this message that mean that your argument isn't valide. Your argument type is {type(argument)}. If you continue to see this message, then, please check your passed arguments.

Maybe we could rename "argument" as "exceptions" and rename "argument_name" as "exception_name".
In this context I think that we could speak about exception rather than seeking to be too generalist. That will make the code more human readable, explicit and semantic.

Reply:
I don't know that I agree with having an exception message that long. Makes it seem like the information should be in the client docstring (both the check method and the constructor) instead.

I agree that the wording could perhaps be better, and I also see that the exception message doesn't actually match the types we check for.

My reasoning for the variable naming is like so:
- `argument_name` is the name of the argument we're currently checking. In itself it doesn't have anything to do with an exception, it's just for giving sensible `ValueError` messages, it will tell whoever catches it exactly which argument (`retry_for` or `do_not_retry_for`) contained an invalid value.
- `argument` is the value we're checking. I'm renaming it to `argument_value` since this is more explicit.

Done in https://github.com/martinnj/pymemcache/commit/6bd21d0261f57d54c8c3d1211088bc27f07f5ee3

Reply:
No problem, the current version LGTM, that was more a suggestion. Leave that as it is if you think that it's ok.